frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -1211,10 +1324,12 @@ pub mod pallet {
             reward_destination: RewardDestination,
             dapp_state: DAppState,
             latest_staked_value: BalanceOf<T>,
+            delegate_account: Option<T::AccountId>,
         ) -> bool {
             reward_destination == RewardDestination::StakeBalance
                 && dapp_state == DAppState::Registered
                 && latest_staked_value > Zero::zero()
+                && delegate_account.is_none()
```

Comment:
It seems a bit misleading to read from storage that staker has set reward destination to _restake_, but restake doesn't happen when claiming the reward.

What is your opinion on this?

Reply:
Well, in the case when freebalance reward destination is set, the same happens. I mean, a read from storage is perfomed and no restaking is done. But I agree that the whole restaking logic could be done better, It doesn´t feel right that almost all claim_staker logic depends on "should restake" conditionals. But I would need to give it more time to think about a alternative implementation.